packages/inter-protocol/src/auction/auctioneer.js

```diff
@@ -384,6 +384,14 @@ export const distributeProportionalSharesWithLimits = (
   return transfers;
 };
 
+// auctionBook changed to create a sub-baggage for each book (to distinguish
+// their quoteNotifier flags) so older auctions will not be upgradable to this
+// version. We believe this version is saving all the necesary state to be
+// upgraded, but that hasn't been verified, so we don't mark it as `canUpgrade'.
```

Comment:
My understanding of those `meta` flags is that the first sentence ("older versions will not be upgradable to this version") is the reason we don't mark this as `canUpgrade`. The second sentence ("saving all the necessary state") is what justifies our use of `canBeUpgraded`. The fact that we haven't verified it is either good note to add but doesn't change the flags, or a reason to not use `canBeUpgraded`.

My opinion is that `canBeUpgraded` is appropriate, but I don't know how much of a stickler we are for "there must be tests to prove this claim before we're allowed to make it".

So maybe add ", so we do not mark this as `canUpgrade`" to the first sentence, and maybe make the second sentence be like "We believe.., so we mark this as `canBeUpgraded` (but this hasn't been verified with an a3p test)".

Reply:
>  The fact that we haven't verified it is either good note to add but doesn't change the flags, or a reason to not use canBeUpgraded.

If we had verified it, it would be 'canUpgrade`. The only time it should be `canBeUpgraded` is when we hope it's upgraded, but haven't proven it.

Reply:
I thought `canUpgrade` is backwards-looking, while `canBeUpgraded` is forward-looking, and that there was no way to express "probably upgradable but not tested".

If I understand it correctly, `canUpgrade` means that this code is capable of upgrading a previous version, which we know is false here because the sub-baggage we added.. (actually we just changed that, we're no longer modifying the way baggage is used, so maybe this *could* upgrade the previous version, but I imagine there are other reasons that might not work).

Reply:
I never did know the difference. Ooh! we have [docs](https://github.com/Agoric/agoric-sdk/blob/d6a7ffb6290c63287c4e1fda66d188440105c2a7/packages/zoe/src/contractFacet/types-ambient.d.ts#L234C1-L235C107) now:

```js
   * - `canUpgrade` means this code can perform an upgrade
   * - `canBeUpgraded` means that the contract stores kinds durably such that the next version can upgrade
```

So I concur that `canBeUpgraded` is correct and that any test for `canUpgrade` from the previous version is certain to fail.